metadata-ingestion/src/datahub/ingestion/source/sac/sac.py

```diff
@@ -359,73 +274,110 @@ def get_workunits_internal(self) -> Iterable[MetadataWorkUnit]:
 
                     yield from self.get_model_workunits(dataset_urn, resource_model)
 
-                mcp = MetadataChangeProposalWrapper(
-                    entityType="dashboard",
-                    changeType=ChangeTypeClass.UPSERT,
-                    entityUrn=dashboard_urn,
-                    aspectName="dashboardInfo",
-                    aspect=DashboardInfoClass(
-                        title=resource.name,
-                        description=resource.description,
-                        lastModified=ChangeAuditStampsClass(
-                            created=AuditStampClass(
-                                time=round(resource.created_time.timestamp() * 1000),
-                                actor=make_user_urn(resource.created_by)
-                                if resource.created_by
-                                else "urn:li:corpuser:unknown",
-                            ),
-                            lastModified=AuditStampClass(
-                                time=round(resource.modified_time.timestamp() * 1000),
-                                actor=make_user_urn(resource.modified_by)
-                                if resource.modified_by
-                                else "urn:li:corpuser:unknown",
-                            ),
-                        ),
-                        customProperties={
-                            "resourceType": resource.resource_type,
-                            "resourceSubtype": resource.resource_subtype,
-                            "storyId": resource.story_id,
-                            "isMobile": str(resource.is_mobile),
-                        },
-                        datasets=sorted(datasets) if datasets else None,
-                        externalUrl=f"{self.config.tenant_url}{resource.open_url}",
+                yield from self.get_resource_workunits(resource, datasets)
+
+    def get_report(self) -> SACSourceReport:
+        return self.report
+
+    def get_resource_workunits(
+        self, resource: Resource, datasets: List[str]
+    ) -> Iterable[MetadataWorkUnit]:
+        dashboard_urn = make_dashboard_urn(
+            platform=self.platform,
+            name=resource.resource_id,
+            platform_instance=self.config.platform_instance,
+        )
+
+        if resource.ancestor_path:
+            mcp = MetadataChangeProposalWrapper(
+                entityUrn=dashboard_urn,
+                aspect=BrowsePathsClass(
+                    paths=[
+                        f"/{self.platform}/{resource.ancestor_path}",
+                    ],
+                ),
+            )
+
+            yield mcp.as_workunit()
+
+            mcp = MetadataChangeProposalWrapper(
+                entityUrn=dashboard_urn,
+                aspect=BrowsePathsV2Class(
+                    path=[
+                        BrowsePathEntryClass(id=folder_name)
+                        for folder_name in resource.ancestor_path.split("/")
```

Comment:
is there any danger of folder names having `/` in them?

Reply:
```/``` are replaced with ```%2F```, when the folders which are returned by the API are evaluated:

https://github.com/datahub-project/datahub/blob/74e0c7f74d5152468f12f97390787daf18bd4e97/metadata-ingestion/src/datahub/ingestion/source/sac/sac.py#L620-L623

The replacement was originally in there because of the browse paths v1, which are still ingested along the browse paths v2 - do we need the v1 browse paths anymore or should I remove them? - ...in the old browse paths the slashes were used to separate the levels, I don't think the replacement is needed for v2 anymore...? However for the config.folder_pattern it's probably easier if the folders are joined using the slash inkl. the replacement of slashes in folder names...:/

Reply:
We don't really need v1 browse paths anymore, but I agree that having this around makes `folder_pattern` simpler so it's fine to leave it in